src/python/pants/backend/project_info/list_roots.py

```diff
@@ -32,19 +32,42 @@ async def all_roots(source_root_config: SourceRootConfig) -> AllSourceRoots:
     source_root_pattern_matcher = source_root_config.get_pattern_matcher()
 
     # Create globs corresponding to all source root patterns.
-    all_paths: Set[str] = set()
+    pattern_matches: Set[str] = set()
     for path in source_root_pattern_matcher.get_patterns():
         if path == "/":
-            all_paths.add("**")
+            pattern_matches.add("**")
         elif path.startswith("/"):
-            all_paths.add(f"{path[1:]}/")
+            pattern_matches.add(f"{path[1:]}/")
         else:
-            all_paths.add(f"**/{path}/")
+            pattern_matches.add(f"**/{path}/")
+
+    # Create globs for any marker files.
+    marker_file_matches: Set[str] = set()
+    for marker_filename in source_root_config.options.marker_filenames:
+        marker_file_matches.add(f"**/{marker_filename}")
 
     # Match the patterns against actual files, to find the roots that actually exist.
-    snapshot = await Get[Snapshot](PathGlobs(globs=sorted(all_paths)))
+    snapshots = await MultiGet(
```

Comment:
I think you can remove line 54 by having this be `pattern_snapshot, marker_file_snapshot = await MultiGet`

Also note that you do not need parentheses.